### PR TITLE
PVR API cleanups

### PR DESCRIPTION
**Pull request by pcercuei**
A small cleanup of the PVR API, with changes that should not break the API nor the ABI:
- Constify data source pointers for the DMA functions;
- use an enum for the palette format;
- use an enum for the DMA transfer type.

### CODE REVIEW

File: kernel/arch/dreamcast/hardware/pvr/pvr_texture.c [hunk 1]
```diff
@@ -20,12 +20,12 @@
 */
 
 /* Load raw texture data from an SH-4 buffer into PVR RAM */
-void pvr_txr_load(void * src, pvr_ptr_t dst, uint32 count) {
+void pvr_txr_load(const void *src, pvr_ptr_t dst, uint32 count) {
```
Comment (andressbarajas) [inline kernel/arch/dreamcast/hardware/pvr/pvr_texture.c:23]:
```suggestion
void pvr_txr_load(const void *src, pvr_ptr_t dst, uint32_t count) {
```

File: kernel/arch/dreamcast/hardware/pvr/pvr_texture.c [hunk 2]
```diff
@@ -52,7 +52,7 @@ void pvr_txr_load(void * src, pvr_ptr_t dst, uint32 count) {
        PVR_TXRLOAD_INVERT
 
 */
-void pvr_txr_load_ex(void * src, pvr_ptr_t dst, uint32 w, uint32 h,
+void pvr_txr_load_ex(const void *src, pvr_ptr_t dst, uint32 w, uint32 h,
```
Comment (andressbarajas) [inline kernel/arch/dreamcast/hardware/pvr/pvr_texture.c:55]:
```suggestion
void pvr_txr_load_ex(const void *src, pvr_ptr_t dst, uint32_t w, uint32_t h,
```

File: kernel/arch/dreamcast/hardware/pvr/pvr_texture.c [hunk 3]
```diff
@@ -143,7 +143,7 @@ void pvr_txr_load_ex(void * src, pvr_ptr_t dst, uint32 w, uint32 h,
 }
 
 /* Load a KOS Platform Independent Image (subject to restraint checking) */
-void pvr_txr_load_kimg(kos_img_t *img, pvr_ptr_t dst, uint32 flags) {
+void pvr_txr_load_kimg(const kos_img_t *img, pvr_ptr_t dst, uint32 flags) {
```
Comment (andressbarajas) [inline kernel/arch/dreamcast/hardware/pvr/pvr_texture.c:146]:
```suggestion
void pvr_txr_load_kimg(const kos_img_t *img, pvr_ptr_t dst, uint32_t flags) {
```

**Comment by andressbarajas** [comment]
Can you typedef your enums to match the KOS coding style? Also eventually we want to move from uint32 => uint32_t so that is why I did a couple reviews. Not required right now so feel free to ignore but it will be done eventually and if we can take some out now, why not?

**Comment by pcercuei** [comment]
I can add the typedefs. For the uint32 -> uint32_t change I think it should be done all at once, and not in unrelated commits.

**Comment by pcercuei** [comment]
@andressbarajas Updated.

Note that I noticed that there's already #539, so I can close my PR if the plan is to merge @gyrovorbis' one.

**Review by QuzarDC — APPROVED** [review]

**Comment by QuzarDC** [comment]
I know there's the gigantic API refactor #539 queued still, but this is useful as it's of a small scope and reduces the number of changes that are piled together in the refactor PR.

**Comment by gyrovorbis** [comment]
> I know there's the gigantic API refactor #539 queued still, but this is useful as it's of a small scope and reduces the number of changes that are piled together in the refactor PR.

Sounds good. I'll handle rebasing the refactor on this one.

**Pull request merged**